lib/wallet/mnemonic.rs

```diff
@@ -24,6 +25,22 @@ pub(crate) fn new_mnemonic() -> Result<Mnemonic, miette::Report> {
     Mnemonic::parse(words).into_diagnostic()
 }
 
+/// Read and parse a mnemonic from a file containing exactly 12 space-separated words.
+pub fn new_from_file(path: impl AsRef<Path>) -> Result<Mnemonic, miette::Report> {
+    let mnemonic_str = std::fs::read_to_string(path)
+        .map_err(|err| miette!("failed to read mnemonic file: {}", err))?;
+
+    let words: Vec<&str> = mnemonic_str.split_whitespace().collect();
+    if words.len() != 12 {
+        return Err(miette!(
+            "mnemonic must be exactly 12 words, got {}",
+            words.len()
+        ));
+    }
```

Comment:
Is this a hard requirement from our side? Validity of the mnemonic is verified on line 41, and valid mnemonics can be of various lengths